Gruntfile.js

```diff
@@ -78,18 +103,56 @@ module.exports = function(grunt) {
       }
     },
 
+    concat: {
+      options: {
+        separator: '\n',
+        // Replace all 'use strict' statements in the code with a single one at the top
+        banner: [
+          '/*',
+          ' * #define CSS_LAYOUT_IMPLEMENTATION',
+          ' * before you include this file in *one* C or C++ file to create the implementation.',
+          ' */\n'
+        ].join('\n'),
+        process: function(src, filepath) {
+          if (path.extname(filepath) === '.c') {
+            return [
+              '#ifdef CSS_LAYOUT_IMPLEMENTATION',
+              src,
+              '#endif // CSS_LAYOUT_IMPLEMENTATION'
+            ].join('\n')
+          }
+          else {
+            return src;
+          }
+        },
+      },
+      dist: {
+        src: ['<%= config.srcFolder %>/Layout.h', '<%= config.srcFolder %>/Layout.c'],
+        dest: '<%= config.distFolder %>/css-layout.h',
+      },
+    },
+
     shell: {
       cCompile: {
-        command: 'gcc -std=c99 -Werror -Wno-padded <%= config.cTestFiles %> -lm -o "./<%= config.cTestOutput %>"'
+        command: cTestCompile
       },
       cTestExecute: {
-        command: './<%= config.cTestOutput %>'
+        command: cTestExecute
       },
       javaCompile: {
-        command: 'javac -cp <%= config.javaLibFolder %>/junit4.jar:<%= config.javaLibFolder %>/jsr305.jar:<%= config.javaLibFolder %>/infer-annotations-1.4.jar -sourcepath ./src/java/src:./src/java/tests <%= config.javaSource %>'
+        command: 'javac -cp <%= config.javaLibFolder %>/junit4.jar' +
```

Comment:
rather than mixing templates strings and concatenation, can you place `path.delimiter`onto the config object and just use templating?